build-scripts/compile_profiles.py

```diff
@@ -0,0 +1,101 @@
+from __future__ import print_function
+
+import argparse
+import sys
+from glob import glob
+
+import ssg.build_yaml
+
+
+class ResolvableProfile(ssg.build_yaml.Profile):
+    def __init__(self, * args, ** kwargs):
+        super(ResolvableProfile, self).__init__(* args, ** kwargs)
+        self.raw_selections = set()
+        self.resolved = False
+
+    def resolve(self, all_profiles):
```

Comment:
This should take `refine_rules` into account as well.

Reply:
Good point, what exactly is this refining? The `refine_rules` dictionary is available at that time, but I don't know what it does. How would this refinement affect profile compilation?
Refinement is effective when a rule is saved as an XML, i.e. later in the process.

Reply:
> Good point, what exactly is this refining? The `refine_rules` dictionary is available at that time, but I don't know what it does. How would this refinement affect profile compilation?

I believe it would behave the same variables.
If profile B extends profile A, and profile B overrides a rule refinement from Profile A, the refinement from Profile B prevails.

For example, `ospp` profile refines the severity of two rules and set them as informational. A profile extending `ospp` could re-refine the rule and say they are not informational.

> Refinement is effective when a rule is saved as an XML, i.e. later in the process.

This applies for variables too, right?

Reply:
I see, so we basically need to do the same as with variables, while making sure that the "selected" yaml key gets the list that contains proper refinements.